apm-agent-plugins/apm-dubbo-plugin/src/main/java/co/elastic/apm/agent/dubbo/helper/DubboTraceHelper.java

```diff
@@ -0,0 +1,166 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.dubbo.helper;
+
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.configuration.CoreConfiguration;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.context.Destination;
+import co.elastic.apm.agent.impl.transaction.AbstractSpan;
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+import co.elastic.apm.agent.impl.transaction.Transaction;
+
+import java.net.InetSocketAddress;
+import java.net.URL;
+import java.security.CodeSource;
+import java.security.ProtectionDomain;
+
+@VisibleForAdvice
+public class DubboTraceHelper {
+
+    @VisibleForAdvice
+    public static ElasticApmTracer tracer;
+
+    private static final String EXTERNAL_TYPE = "external";
+
+    private static final String DUBBO_SUBTYPE = "dubbo";
+
+    public static final String SPAN_KEY = "span";
+
+    public static void init(ElasticApmTracer tracer) {
+        DubboTraceHelper.tracer = tracer;
+    }
+
+    @VisibleForAdvice
+    public static Span createConsumerSpan(Class<?> apiClass, String methodName, Class<?>[] paramClasses,
+                                          String version, InetSocketAddress remoteAddress) {
+        TraceContextHolder<?> traceContext = DubboTraceHelper.tracer.getActive();
+        if (traceContext == null) {
+            return null;
+        }
+        Span span = traceContext.createExitSpan();
+        if (span == null) {
+            return null;
+        }
+
+        span.withType(EXTERNAL_TYPE)
+            .withSubtype(DUBBO_SUBTYPE);
+        fillName(span, apiClass, methodName, paramClasses, version);
+
+        Destination destination = span.getContext().getDestination();
+        destination.withAddress(remoteAddress.getHostName()).withPort(remoteAddress.getPort());
+
+        Destination.Service service = destination.getService();
+        service.withType(EXTERNAL_TYPE).withResource(DUBBO_SUBTYPE).withName(DUBBO_SUBTYPE);
+
+        return span.activate();
+    }
+
+    private static void fillName(AbstractSpan<?> span, Class<?> apiClass, String methodName,
+                                 Class<?>[] paramClasses, String version) {
+        span.appendToName(apiClass.getName())
```

Comment:
Can we make the name just `"${simpleClassName}#${methodName}"`.

This would make it more consistent how we name transactions for other technologies like Spring MVC and JAS-RS

Reply:
Use `java.lang.StringBuilder#append(java.lang.CharSequence, int, int)` instead of `Class#getSimpleClassName` for that.

Reply:
what about "${simpleClassName}#${methodName}(${parameters})", because ${parameters} is a unique part of dubbo API